R/AllClasses.R

```diff
@@ -28,3 +28,5 @@ setClass("LinearEmbeddingMatrix",
                    NAMES = "character_OR_NULL",
                    factorData = "DataFrame"),
          contains = "Annotated")
+
+setClass("SummarizedExperimentByColumn", slots=c(se="SummarizedExperiment"))
```

Comment:
Don't really mind either way about **1**, the user should never see this. I named it as it is to avoid having to think about multiple layers of transposition and re-transposition; now we only have to know that the subsetting of this class is parallel to the columns, and that's that.

**2**... sounds hairy, it could break a whole lot of stuff under the hood - internal _S4Vectors_ functions use the various generics as well. Besides, there's no benefit because we don't want to be returning this... thing... to the users, especially if the length and subsetting semantics are all messed up. We'd have to "purify" the SE to return it anyway.

Reply:
I suspect you are right about 2...and if we do have a way to "untranspose" it for any calculations that callees will need then you are right, no need to try to use inheritance.